packages/core/src/logger/Logger.spec.ts

```diff
@@ -0,0 +1,116 @@
+import 'reflect-metadata';
+import { Logger, LogLevel } from './Logger';
+
+const logLevels = [
+  LogLevel.VERBOSE,
+  LogLevel.NOTICE,
+  LogLevel.WARN,
+  LogLevel.ERROR,
+  LogLevel.SILENT
+];
+
+const getLogLevelMethodName = (
+  logLevel: LogLevel
+): 'debug' | 'log' | 'warn' | 'error' | undefined => {
+  switch (logLevel) {
+    case LogLevel.VERBOSE:
+      return 'debug';
+    case LogLevel.NOTICE:
+      return 'log';
+    case LogLevel.WARN:
+      return 'warn';
+    case LogLevel.ERROR:
+      return 'error';
+  }
+
+  return undefined;
+};
+
+const getLogLevelName = (logLevel: LogLevel): string => {
+  switch (logLevel) {
+    case LogLevel.VERBOSE:
+      return 'VERBOSE';
+    case LogLevel.NOTICE:
+      return 'NOTICE';
+    case LogLevel.WARN:
+      return 'WARN';
+    case LogLevel.ERROR:
+      return 'ERROR';
+    case LogLevel.SILENT:
+      return 'SILENT';
+  }
+};
+
+describe('Logger', () => {
+  const mockedConsole = jest.spyOn(console, 'log');
+
+  afterEach(() => jest.resetAllMocks());
+
+  describe('level config', () => {
+    it('default level is NOTICE', () => {
+      const logger = new Logger();
+      expect(logger.logLevel).toEqual(LogLevel.NOTICE);
+    });
+
+    it('level can be set via constructor', () => {
+      const logger = new Logger(LogLevel.ERROR);
+      expect(logger.logLevel).toEqual(LogLevel.ERROR);
+    });
+
+    it('level can be changed via setter', () => {
+      const logger = new Logger();
+      logger.logLevel = LogLevel.WARN;
+      expect(logger.logLevel).toEqual(LogLevel.WARN);
+    });
+  });
+
+  describe('logging', () => {
```

Comment:
Considering the comment above, let's rewrite these tests to make them more obvious. Moreover, It is better to use `.each` to set up test data, avoiding any logical conditions